### ISSUE TITLE
home button does not restore initial rotation

### ISSUE THREAD

**Issue by nsg21** [openseadragon/openseadragon#2039]
I would expect home button to not just set initial zoom and position, but also a rotation (specified in .degrees). As it turns out it does not. Is it by design?
(For context: I have .rotationIncrement set to 15 and use it to view scanned maps which have labels and notes in each and every orientation. To return to initial view to zoom in on a next part I have to press home and then multiple rotate button until it rotates correct side up. Then I zoom in on the area of interest and rotate it until all the labels are in required orientation.)

**Comment by iangilman** [comment]
Good point! I think we just didn't think about it when rotation was implemented. It seems like a decent feature addition, though it might break apps that start off with a non-standard rotation. I suppose the best would be if hitting the home button returned the rotation to whatever was initially set via the viewer options. 

If you're interested in implementing this feature, I can point you to the right part of the code.

**Comment by nsg21** [comment]
Yes, thank you. I would appreciate any pointer you can give.

**Comment by iangilman** [comment]
Great! You'll want to add a call to `this.setRotation` inside of `Viewport.goHome`:

https://github.com/openseadragon/openseadragon/blob/f097808c95ea11ed8ac9cd57044d98ce3da38914
/src/viewport.js#L269

I think it should happen before the `fitBounds` in that function, since the rotation may affect the fitBounds.

It should be based on the `degrees` value that was originally given to the viewport on construction, so you'll need to save that value in the constructor for later use. 

Also, for efficiency, it would be good to check if the current `degrees` is different from the original degrees, and if they're the same, don't bother changing the rotation.
